homeassistant/components/logbook/models.py

```diff
@@ -55,17 +60,27 @@ def __init__(
             self.data = event_data
         else:
             self.data = self._event_data_cache[source] = cast(
-                dict[str, Any], json.loads(source)
+                dict[str, Any], json_loads(source)
             )
 
+    @property
+    def context_id(self) -> str | None:
+        """Return the context id."""
+        return bytes_to_ulid_or_none(self.context_id_bin)
+
+    @property
+    def context_parent_id(self) -> str | None:
+        """Return the context parent id."""
+        return bytes_to_ulid_or_none(self.context_parent_id_bin)
```

Comment:
These are not heavily optimized but currently only the script and automation platform return them in the payload so lazy properties seemed like the best here since most queries will only call this 3-4 times.